### PR TITLE
Improve testing setup and add travis.yml

### PR DESCRIPTION
**Pull request by blasterpal**
* Goal is to add more tests on agent
  * Also have CI

### CODE REVIEW

**Comment by blasterpal** [comment]
@randy-girard please run the tracer_spec.rb and let me know if you are seeing failures. I am:

```
Failures:

  1) AppPerfRpm trace header is set should trace with that trace key
     Failure/Error: ::AppPerfRpm.store(event)

       AppPerfRpm received :store with unexpected arguments
         expected: (["third", 1, 0.0, 0.0, {}])
              got: (["third", "23b54169ece2a6b531ce7ceef59408a50d683948", 0.0, 0.0, {}])
     # ./lib/app_perf_rpm/tracer.rb:55:in `trace'
     # ./spec/app_perf_rpm/tracer_spec.rb:16:in `block (5 levels) in <top (required)>'
     # ./lib/app_perf_rpm/tracer.rb:51:in `trace'
     # ./spec/app_perf_rpm/tracer_spec.rb:15:in `block (4 levels) in <top (required)>'
     # ./lib/app_perf_rpm/tracer.rb:37:in `block in start_trace'
     # ./lib/app_perf_rpm/tracer.rb:51:in `trace'
     # ./lib/app_perf_rpm/tracer.rb:36:in `start_trace'
     # ./spec/app_perf_rpm/tracer_spec.rb:14:in `block (3 levels) in <top (required)>'

  2) AppPerfRpm when sample rate is 100 should trace through all traces
     Failure/Error: ::AppPerfRpm.store(event)

       AppPerfRpm received :store with unexpected arguments
         expected: (["third", 1, 0.0, 0.0, {}])
              got: (["third", "71f9dadaf5373a417de6a7670bd8c5e47b9a2490", 0.0, 0.0, {}])
     # ./lib/app_perf_rpm/tracer.rb:55:in `trace'
     # ./spec/app_perf_rpm/tracer_spec.rb:44:in `block (5 levels) in <top (required)>'
     # ./lib/app_perf_rpm/tracer.rb:51:in `trace'
     # ./spec/app_perf_rpm/tracer_spec.rb:43:in `block (4 levels) in <top (required)>'
     # ./lib/app_perf_rpm/tracer.rb:37:in `block in start_trace'
     # ./lib/app_perf_rpm/tracer.rb:51:in `trace'
     # ./lib/app_perf_rpm/tracer.rb:36:in `start_trace'
     # ./spec/app_perf_rpm/tracer_spec.rb:42:in `block (3 levels) in <top (required)>'

Finished in 0.04249 seconds (files took 0.11829 seconds to load)
5 examples, 2 failures

Failed examples:

rspec ./spec/app_perf_rpm/tracer_spec.rb:8 # AppPerfRpm trace header is set should trace with that trace key
rspec ./spec/app_perf_rpm/tracer_spec.rb:36 # AppPerfRpm when sample rate is 100 should trace through all traces
```

**Comment by randy-girard** [comment]
Rebase the branch. I just fixed the code.

File: spec/app_perf_rpm_spec.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require 'spec_helper'
+require 'socket'
```
Comment (randy-girard) [inline spec/app_perf_rpm_spec.rb]: Why is this needed?

Comment (blasterpal) [inline spec/app_perf_rpm_spec.rb]: I was receiving this error running just that spec: 

```
hbeaver~/code/vendor/app_perf_rpm (improve_testing)$ be rspec spec/app_perf_rpm_spec.rb

Randomized with seed 12788
.F...

Failures:

  1) AppPerfRpm#host returns the host name
     Failure/Error: expect(Socket).to receive(:gethostname) { "host" }

     NameError:
       uninitialized constant Socket
     # ./spec/app_perf_rpm_spec.rb:10:in `block (3 levels) in <top (required)>'

Finished in 0.01044 seconds (files took 0.16756 seconds to load)
5 examples, 1 failure

Failed examples:

rspec ./spec/app_perf_rpm_spec.rb:9 # AppPerfRpm#host returns the host name

Randomized with seed 12788
```

It doesn't appear to be order dependent for specs when executing suite (which is a reason I added `:random` ordering to RSpec config. 

My current system:
Ruby 2.2.2p95
MaxOS 10.12.4
